pkg/rehearse/jobs.go

```diff
@@ -821,29 +832,51 @@ func (e *Executor) waitForJobs(jobs sets.String, selector string) (bool, error)
 	}
 }
 
-func removeConfigResolverFlags(args []string) []string {
-	newArgs := []string{}
-	ignoreNext := false
-	for _, arg := range args {
-		if ignoreNext {
-			ignoreNext = false
-			continue
-		}
-		ignore := false
-		for _, ignoredArg := range []string{"-resolver-address", "-org", "-repo", "-branch", "-variant"} {
-			// Handle both single and double dash forms of go flags
-			if strings.Contains(arg, ignoredArg) {
-				ignore = true
+func removeConfigResolverFlags(args []string) ([]string, config.Info) {
+	var usedConfig config.Info
+	toConfig := map[string]*string{
+		"org":     &usedConfig.Org,
+		"repo":    &usedConfig.Repo,
+		"branch":  &usedConfig.Branch,
+		"variant": &usedConfig.Variant,
+	}
+	var newArgs []string
+	for len(args) > 0 {
```

Comment:
err...What about `for idx := range args { current := args[idx]` to allow understanding this in less than 30 seconds?

Reply:
The need to eat an additional arg for `--arg value` forms in the middle of the block makes simple iteration messy, you need to add hacks like `skipNext = true`, which I think result in even less readable code.

Reply:
I think having code in a loop body that mutates the iterable that is being looped over is pretty unexpected and hence hard to read

Reply:
We will ask new hires to explain this method for their interview

Reply:
It's a well-known pattern in parsing (this use case is similar), you consume a "stream" of tokens and process them... I think even most of actual parameter parsers work exactly like this (for exactly the same reason).

Do you think adding a comment that explains why a strange loop is used would help, or are you adamant in that the iteration should be your way?

Reply:
the consuming stream part yes, the manipulating stream during process no

Reply:
But consuming is exactly what the loop does, it consumes items from the beginning of the slice until it is empty. Some iterations consume a single item, some consume two. There's no more manipulation than that.